source/megengine.js

```diff
@@ -688,11 +693,7 @@ megengine.TensorType = class {
             this._dataTypeName = megengine.Utility.enum(megengine.schema, dtype_class, this._value).toLowerCase();
         }
         megengine.TensorType._dataTypeMap = megengine.TensorType._dataTypeMap || new Map([
-            [ 'bool', 'boolean' ],
-            [ 'byte', 'uint8' ], [ 'quantizeds4asymm', 'uint8' ], [ 'quantizeds8asymm', 'uint8' ], [ 'uintb4', 'uint8' ],
-            [ 'quantizeds1', 'int8' ], [ 'quantizeds4', 'int8' ], [ 'quantizeds8', 'int8' ], [ 'intb1', 'int8' ], [ 'intb2', 'int8' ], [ 'intb4', 'int8' ], [ 'qint8', 'int8' ],
-            [ 'quantizeds16', 'int16' ],
-            [ 'quantizeds32', 'int32' ]
+            [ 'bool', 'boolean' ], [ 'byte', 'uint8' ], [ 'quantizeds4asymm', 'uint8' ], [ 'quantizeds8asymm', 'uint8' ], [ 'uintb4', 'uint8' ], [ 'quantizeds1', 'int8' ], [ 'quantizeds4', 'int8' ], [ 'quantizeds8', 'int8' ], [ 'intb1', 'int8' ], [ 'intb2', 'int8' ], [ 'intb4', 'int8' ], [ 'qint8', 'int8' ], [ 'quantizeds16', 'int16' ], [ 'quantizeds32', 'int32' ], [ 'TypePlaceHolder', 'boolean' ]
```

Comment:
Each pair on new line to make diff easier in the future?

Reply:
Good idea, done.

Reply:
Why is `?` mapped to `boolean` via `[ 'bool', 'boolean' ], [ '?', 'boolean' ]`? Also `this._dataTypeName` and `this._dataType` should be identical so `toString()` uses the same name as `type.dataType`?

Reply:
After testing, `['?', 'boolean']` here can be deleted.
The `this._dataType` here is mainly used to dynamically read the numerical data of the `megengine.Tensor` during visualization. It seems that the reading of the numerical data of the `*.Tensor` is no longer customized by the .js files of various frameworks? Therefore, `this._dataTypeName` is used to display the data type of mge.

Reply:
Is there a way to generalize this? For example, can these types be added and supported for other formats as well by the shared rendering code?

Reply:
The mapping here only involves some special data types (such as quantizeds4asymm), but there is no mapping for common data types (such as float32). For mge models in other formats (such as .tm format), the special data types they support are also included in this correspondence. So don't generalize? I wonder if I understand what you mean correctly?

Reply:
Instead of mapping `quantizeds4asymm` to `uint8`, is `quantizeds4asymm` a common data type that is supported by other frameworks? What would be the right name to use for all frameworks? Should `dialog.Tensor` support these data types? For example, there is `int4' in other frameworks, if it is not encoded in a standard way maybe it should report the standard name and not return data for rendering?

Reply:
Data type `quantizeds4asymm` is defined by us for quantifying data types, so it is not an common data type supported by other frameworks. Its corresponding name in `dialog.Tensor` should be `quint8`, which is also the name used by other frameworks. done.